lib/data_table.rb

```diff
@@ -760,19 +760,16 @@ def data_field_for(name)
       def to_json(view_context, params = {})
         resultset, total_count = get_resultset_and_count(params)
 
-        if params.key?(:dataField)
-          request_column = self.columns.detect{ |c| c.name == params[:dataField].split('.').last.to_sym }
-          Jbuilder.encode do |json|
-            json.items(resultset) do |instance|
-              value = request_column.text(instance, view_context) rescue nil
-              json.set! :key, value
-            end
-          end
-        else
-          Jbuilder.encode do |json|
-
-            json.items(resultset) do |instance|
-
+        group_params = params.fetch('groupOptions', {})
+        group_params = JSON.parse(group_params) if group_params.is_a?(String) && group_params.present?
+
+        Jbuilder.encode do |json|
+          json.items(resultset) do |instance|
+            if !params.key?(:skip) && !params.key?(:take) && group_params.size == 1 # if lookup filter row
```

Comment:
Please add a comment explaining what this is for and why it is here. Also, why are you using `skip` and `take`? Is that the only way to identify that this is from the filter row?

Also, I don't like the fact that the params check is inside the loop. This adds unnecessary compute to this loop. Please move those outside the loop. Duplicate if need be.